### PR TITLE
Return annotation brackets

### LINKED ISSUE #2041: Redundant parentheses around return annotation
```python
def infer_encoding_trajectory(encoding: EncodingElement,
                              observation: PoolingMessage,
                              prediction: PredictionMessage,
                              weights: hk.Params,
                              rng: Generator,
                              iterations: int) -> (
                                  tuple[EncodingState, EncodingTrajectory]):
```
became
```python
def infer_encoding_trajectory(
    encoding: EncodingElement,
    observation: PoolingMessage,
    prediction: PredictionMessage,
    weights: hk.Params,
    rng: Generator,
    iterations: int,
) -> (tuple[EncodingState, EncodingTrajectory]):
```
The parentheses were only there to allow wrapping and should have been removed.  In general, black doesn't need to preserve parentheses around an entire expression, including a type annotation.

### LINKED ISSUE #2699: Line too long: String in return annotation
**To Reproduce**

Given input:
```python
class ThisIsTrulyUnreasonablyExtremelyLongClassName:
    pass


def frobnicate() -> "ThisIsTrulyUnreasonablyExtremelyLongClassName | list[ThisIsTrulyUnreasonablyExtremelyLongClassName]":
    pass
```

Black does not split the long type hint of the function return type

Even if I explicitly attempt to split the string:
```python
def frobnicate() -> "ThisIsTrulyUnreasonablyExtremelyLongClassName | " \
                    "list[ThisIsTrulyUnreasonablyExtremelyLongClassName]":
    pass
```

then Black collapses it back together :(
```python
def frobnicate() -> "ThisIsTrulyUnreasonablyExtremelyLongClassName | " "list[ThisIsTrulyUnreasonablyExtremelyLongClassName]":
    pass
```

**Workaround**

Only by parenthesizing the type hint I can convince Black to keep it reasonable.

```python
def frobnicate() -> (
    "ThisIsTrulyUnreasonablyExtremelyLongClassName | "
    "list[ThisIsTrulyUnreasonablyExtremelyLongClassName]"
):
    pass
```

**Environment**

- Black's version: 21.12b0
- OS and Python version: macOS, Python 3.10.1

### PR DESCRIPTION
**Pull request by jpy-git**
### Description

Closes #2041 and #2699. Allows us to better control placement of return annotations.

a) removes redundant parens:
```python
def foo() -> (int): ...
```
becomes
```python
def foo() -> int: ...
```

b) moves very long type annotations onto their own line:
```python
def foo() -> intsdfsafafafdfdsasdfsfsdfasdfafdsafdfdsfasdskdsdsfdsafdsafsdfdasfffsfdsfdsafafhdskfhdsfjdslkfdlfsdkjhsdfjkdshfkljds:
    return 2
```
becomes
```python
def foo() -> (
    intsdfsafafafdfdsasdfsfsdfasdfafdsafdfdsfasdskdsdsfdsafdsafsdfdasfffsfdsfdsafafhdskfhdsfjdslkfdlfsdkjhsdfjkdshfkljds
):
    return 2
```

### Checklist - did you ...

- [X] Add a CHANGELOG entry if necessary?
- [X] Add / update tests if necessary?
- [X] Add new / update outdated documentation?

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -14,6 +14,8 @@
 
 <!-- Changes that affect Black's preview style -->
 
+- Optional brackets around return anotations (#2990)
```
Comment (JelleZijlstra) [inline CHANGES.md:17]:
```suggestion
- Parentheses around return annotations are now managed (#2990)
```
Not sure I like the "managed" wording, but we should say what changed

File: tests/data/return_annotation_brackets.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
+# Control
+def double(a: int) -> int:
+    return 2*a
+
+# Remove the brackets
+def double(a: int) -> (int):
```
Comment (JelleZijlstra) [inline tests/data/return_annotation_brackets.py:6]: Also add an example with multiple sets of parens. And maybe for additional fun, what if there is more whitespace between the parentheses?

**Review by JelleZijlstra — COMMENTED** [review]
Thanks! Some small feedback

File: tests/data/return_annotation_brackets.py [hunk 3]
```diff
@@ -53,6 +53,27 @@ def foo(a: int, b: int, c: int,) -> intsdfsafafafdfdsasdfsfsdfasdfafdsafdfdsfasd
 def foo(a: int, b: int, c: int,) -> int:
     return 2
 
+# Deeply nested brackets
+# with *interesting* spacing
```
Comment (JelleZijlstra) [inline tests/data/return_annotation_brackets.py:57]: love it

Comment (jpy-git) [reply tests/data/return_annotation_brackets.py:57]: I pray I never meet a developer who likes to write their brackets like these examples 😅

**Review by JelleZijlstra — APPROVED** [review]

File: tests/data/return_annotation_brackets.py (hunk 3 above)
Comment (JelleZijlstra) [reply tests/data/return_annotation_brackets.py:57]: You missed an opportunity to write ASCII art using weirdly spaced parentheses

Comment (jpy-git) [reply tests/data/return_annotation_brackets.py:57]: I've got a few sad faces `):` in there

**Review by ichard26 — APPROVED** [review]
Excellent work @jpy-git :tada:

**Pull request merged**